pkg/kubelet/nodeshutdown/nodeshutdown_manager_windows_test.go

```diff
@@ -0,0 +1,352 @@
+					// for volume unmount operations that take longer than the allowed grace period.
+					fmt.Errorf("unmount timeout"),
+				),
+				shutdownGracePeriodByPodPriority: []kubeletconfig.ShutdownGracePeriodByPodPriority{
+					{
+						Priority:                   1,
+						ShutdownGracePeriodSeconds: 2,
+					},
+				},
+				getPods: func() []*v1.Pod {
+					return []*v1.Pod{
+						makePod("normal-pod", 1, nil),
+						makePod("critical-pod", 2, nil),
+					}
+				},
+				killPodFunc: func(pod *v1.Pod, isEvicted bool, gracePeriodOverride *int64, fn func(*v1.PodStatus)) error {
+					return nil
+				},
+				syncNodeStatus: syncNodeStatus,
+				clock:          fakeclock,
+			},
+			wantErr:                false,
+			expectedOutputContains: "Failed while waiting for all the volumes belonging to Pods in this group to unmount",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			logger := ktesting.NewLogger(t,
+				ktesting.NewConfig(
+					ktesting.BufferLogs(true),
+				),
+			)
+			m := &managerImpl{
+				logger:                logger,
+				recorder:              tt.fields.recorder,
+				nodeRef:               tt.fields.nodeRef,
+				probeManager:          tt.fields.probeManager,
+				getPods:               tt.fields.getPods,
+				syncNodeStatus:        tt.fields.syncNodeStatus,
+				nodeShuttingDownMutex: sync.Mutex{},
+				nodeShuttingDownNow:   tt.fields.nodeShuttingDownNow,
+				podManager: &podManager{
+					logger:                           logger,
+					volumeManager:                    tt.fields.volumeManager,
+					shutdownGracePeriodByPodPriority: tt.fields.shutdownGracePeriodByPodPriority,
+					killPodFunc:                      tt.fields.killPodFunc,
+					clock:                            tt.fields.clock,
+				},
+			}
+			if err := m.ProcessShutdownEvent(); (err != nil) != tt.wantErr {
+				t.Errorf("managerImpl.processShutdownEvent() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			underlier, ok := logger.GetSink().(ktesting.Underlier)
+			if !ok {
+				t.Fatalf("Should have had a ktesting LogSink, got %T", logger.GetSink())
+			}
+
+			log := underlier.GetBuffer().String()
+			if tt.expectedOutputContains != "" && !strings.Contains(log, tt.expectedOutputContains) {
+				// Log will be shown on failure. To see it
+				// during a successful run use "go test -v".
+				t.Errorf("managerImpl.processShutdownEvent() should have logged %s, see actual output above.", tt.expectedOutputContains)
+			}
+
+			if tt.expectedOutputNotContains != "" && strings.Contains(log, tt.expectedOutputNotContains) {
+				// Log will be shown on failure. To see it
+				// during a successful run use "go test -v".
+				t.Errorf("managerImpl.processShutdownEvent() should have not logged %s, see actual output above.", tt.expectedOutputNotContains)
+			}
+		})
+	}
+}
+
+func Test_addToExistingOrder(t *testing.T) {
```

Comment:
please add re-entry tests. Kubelet restart should not be growing this list and ideally should just keep it as-is.

Reply:
will add one